src/DGtal/io/readers/ITKReader.ih

```diff
@@ -64,7 +64,8 @@ namespace DGtal {
     const Domain& domain = dgtal_itk_image.domain();
 
     typedef ConstImageAdapter<DGtalITKImage, Domain, functors::Identity, Value, Functor> AdaptedImage;
-    const AdaptedImage adapted(dgtal_itk_image, domain, functors::Identity(), aFunctor);
+    const functors::Identity identityFunctor{};
```

Comment:
I apply the same change of @rolanddenis ;) looks fine without with my compilo but I suppose that there a reason.